src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,73 @@
 package core.basesyntax;
 
+import static java.lang.Integer.parseInt;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    public static final String REGEX = " ";
+    public static final String BUY = "buy";
+    public static final String SUPPLY = "supply";
+    public static final String RESULT_REGEX = ",";
+    public static final String RESULT_NAME = "result";
+
+    public static void getStatistic(String fromFileName, String toFileName) {
+        readDataFromFile(fromFileName, toFileName);
+    }
+
+    private static void readDataFromFile(String fromFileName, String toFileName) {
+        File readFile = new File(fromFileName);
 
+        StringBuilder builder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(readFile))) {
+            while (reader.ready()) {
+                builder.append(reader.readLine()).append(REGEX);
+            }
+        } catch (IOException exception) {
+            throw new RuntimeException("Can't read data from file " + readFile, exception);
+        }
+        String formattedData = getFormattedData(builder.toString());
+        writeDataToFile(formattedData, toFileName);
```

Comment:
Return `builder.toString()` from`readDataFromFile` and use `String formattedData = getFormattedData(...);` `writeDataToFile(formattedData, toFileName);` in `getStatistic` method